repl/src/main.rs

```diff
@@ -53,11 +59,33 @@ fn main() -> Result<()> {
     };
     let repl_params = Shared::new(repl_params);
 
+    //Create our MeTTa runtime environment
     let mut metta = MettaShim::new(repl_params.clone());
 
+    //Spawn a signal handler background thread, to deal with passing interrupts to the execution loop
+    let mut signals = Signals::new(&[SIGINT])?;
+    thread::spawn(move || {
+        for _sig in signals.forever() {
+            //Assume SIGINT, since that's the only registered handler
+            let mut signal_state = SIGNAL_STATE.lock().unwrap();
+            match *signal_state {
+                0 => println!("Interrupt received, stopping MeTTa..."),
+                1 => println!("Stopping in progress.  Please wait..."),
+                _ => {
+                    println!("Ok, I get it!  Yeesh!");
+                    exit(-1);
+                },
+            }
+            *signal_state += 1;
+            drop(signal_state);
```

Comment:
No.  In this case it doesn't.  The lock will go out of scope and be dropped here regardless.

I like to explicitly drop locks however, because the default behavior is to drop when the variable goes out of scope, but it's not always obvious where that happens.  With object lifecycle, that's fine.  But accidentally holding a lock too long can be very bad.  So I like to make it obvious where the critical section begins and ends.